### DISCUSSION TITLE
Guide about benmarking JavaScript perf on Hermes

### DISCUSSION THREAD

**Discussion by cometkim** [facebook/hermes#1439] [General]
## Description

I'm doing a microbenchmark for JavaScript libraries. Specifically https://github.com/cometkim/unicode-segmenter/issues/46

But I'm not sure how to do it correctly in Hermes. Is there any entry to sub-milliseconds timings? Or is there any internal or third-party tooling commonly used for tracking JavaScript performance?

**Comment by neildhar** [comment]
Hey @cometkim, we don't have built in support for sub-millisecond timestamps. The most reliable way to take such a measurement would be to repeat your benchmark many times so you get a more stable result, and then divide it by the number of iterations. If you're trying to profile performance, Hermes provides a sampling profiler, but that is also far too coarse grained for sub-millisecond measurements.

**Reply by cometkim** [reply]
Ok then, perhaps I could scale the input size to extream so operations take milliseconds.
